nova_vm/src/ecmascript/execution/environments.rs

```diff
@@ -44,34 +125,136 @@ impl Default for Environments {
     }
 }
 
+/// ### [9.1.2.1 GetIdentifierReference ( env, name, strict )](https://tc39.es/ecma262/#sec-getidentifierreference)
+///
+/// The abstract operation GetIdentifierReference takes arguments env (an
+/// Environment Record or null), name (a String), and strict (a Boolean) and
+/// returns either a normal completion containing a Reference Record or a throw
+/// completion.
+pub(crate) fn get_identifier_reference(
+    agent: &mut Agent,
+    env: Option<EnvironmentIndex>,
+    name: &str,
+    strict: bool,
+) -> JsResult<Reference> {
+    // 1. If env is null, then
+    let Some(env) = env else {
+        // a. Return the Reference Record {
+        return Ok(Reference {
+            // [[Base]]: UNRESOLVABLE,
+            base: Base::Unresolvable,
+            // [[ReferencedName]]: name,
+            referenced_name: ReferencedName::String(name.into()),
+            // [[Strict]]: strict,
+            strict,
+            // [[ThisValue]]: EMPTY
+            this_value: None,
+        });
+        // }.
+    };
+
+    // 2. Let exists be ? env.HasBinding(name).
+    let exists = match env {
```

Comment:
Should we reproduce every Environment table method on `EnvironmentIndex` which would find the respective method to call?